source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -66,34 +79,47 @@ Http::FilterTrailersStatus Http1BridgeFilter::encodeTrailers(Http::ResponseTrail
     chargeStat(trailers);
   }
 
-  if (do_bridging_) {
-    // Here we check for grpc-status. If it's not zero, we change the response code. We assume
-    // that if a reset comes in and we disconnect the HTTP/1.1 client it will raise some type
-    // of exception/error that the response was not complete.
-    const Http::HeaderEntry* grpc_status_header = trailers.GrpcStatus();
-    if (grpc_status_header) {
-      uint64_t grpc_status_code;
-      if (!absl::SimpleAtoi(grpc_status_header->value().getStringView(), &grpc_status_code) ||
-          grpc_status_code != 0) {
-        response_headers_->setStatus(enumToInt(Http::Code::ServiceUnavailable));
-      }
+  if (!do_bridging_) {
```

Comment:
Could this function be simplified?
```suggestion
  if (do_bridging_) {
    doTrailers(trailers, /*set_grpc_status_and_message=*/true);
  }
  return Http::FilterTrailersStatus::Continue;
```